### PR TITLE
client: pay back WRU when the request fails

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5851.

### What is changed and how does it work?

```commit-message
When a write request fails, pay back the WRU cost we count before.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/resource_manager/client/model.go [hunk 1]
```diff
@@ -83,20 +84,44 @@ func (kc *KVCalculator) BeforeKVRequest(consumption *rmpb.Consumption, req Reque
...
-		consumption.RRU += float64(kc.CPUMsCost) * kvCPUMs
+		// For now, we can only collect the KV CPU cost for a read request.
+		kc.calculateCPUCost(consumption, res)
+	} else if !res.Succeed() {
+		// If the write request is not successfully returned, we need to pay back the WRU cost.
+		kc.payBackWriteCost(consumption, req)
 	}
 	// A write request may also read data, which should be counted into the RRU cost.
+	// This part should be counted even if the request does not succeed.
+	kc.calculateReadCost(consumption, res)
```
Comment (BornChanger) [inline client/resource_manager/client/model.go:104]: Just curious, can this logic be leveraged by some one to intentionally issue volume of failing write requests, while no cost at all?

Comment (nolouch) [reply client/resource_manager/client/model.go:104]: Here only pay back the `write bytes` part.

**Review by nolouch — APPROVED** [review]
lgtm

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**